### DISCUSSION TITLE
Use different vector table in PgVectorStore

### DISCUSSION THREAD

**Discussion by luizhsalazar** [spring-projects/spring-ai#569] [Q&A]
Hi guys!

I'm currently developing a RAG project using Spring AI framework. To work with vector database I'm using the following depency (PGVector Store):

```
<dependency>
    <groupId>org.springframework.ai</groupId>
    <artifactId>spring-ai-pgvector-store-spring-boot-starter</artifactId>
</dependency>
```

What I'm struggling is how to use a different table name in my project. As default, the PgVectorStore class is set to work with a specific table (vector_store):

`public static final String VECTOR_TABLE_NAME = "vector_store";`

As I already have a table populated with all my embeddings, I want to update this in order to use this already setup table (which is not called "vector_store"). Am I missing something or there is no way to do this?

The Spring AI version I've been working is 0.8.1.

Thanks in advance!

**Comment by akash4chandran** [comment]
Hi,
I faced a similar challenge while working on a project using the Spring AI framework with the PgVectorStore. By default, PgVectorStore uses the vector_store table, but I needed to use a custom table name.

Here's how I managed to use a custom table within my custom schema:

```
CREATE EXTENSION IF NOT EXISTS vector;
CREATE EXTENSION IF NOT EXISTS hstore;
CREATE EXTENSION IF NOT EXISTS "uuid-ossp";

CREATE SEQUENCE custom_schema.custom_seq
  INCREMENT 1
  MINVALUE 1
  MAXVALUE 9223372036854775807
  START 1
  CACHE 1;

DROP TABLE IF EXISTS custom_schema.custom_table;
CREATE TABLE IF NOT EXISTS custom_schema.custom_table (
    id int8 NOT NULL UNIQUE default NEXTVAL('custom_schema.custom_seq'),
    unique_id UUID NOT null UNIQUE,
    content text,
    metadata json,
    embedding vector(1536),
    created_on TIMESTAMPTZ,
    modified_on TIMESTAMPTZ,
    CONSTRAINT custom_pkey PRIMARY KEY (id)
);
```
**Using Hibernate for Custom Entity**

To work with this custom table using Hibernate, you can use the hibernate-vector dependency to map your custom entity. Here’s how you can configure it:

```
<dependency>
    <groupId>org.hibernate.orm</groupId>
    <artifactId>hibernate-vector</artifactId>
    <version>6.4.0.Final</version>
</dependency>
```
This setup allows you to use your custom table with PgVectorStore and Hibernate. For more details on using and configuring PgVectorStore, refer to [pgvector-java](the https://github.com/pgvector/pgvector-java)

Feel free to reach out if you have any questions or need further assistance!

**Comment by sadhasivam** [comment]
`spring.ai.vectorstore.pgvector.table-name=custom_table ` however this has to follow same schematic convention or some kind of trigger to update back. i am tyring to find out how to get the primary key of this table that got inserted to join with with domain table .
